tests/ra_aid/agents/test_ciayn_agent.py

```
@@ -154,3 +155,49 @@ def test_trim_chat_history_both_limits():
+    ])
+    def test_valid_function_calls(self, test_input):
+        """Test function call patterns that should pass validation."""
+        assert not validate_function_call_pattern(test_input)
+
+    @pytest.mark.parametrize("test_input", [
+        "",
+        "Invalid!function()",
+        "missing_parens",
+        "unmatched(parens))",
+        "multiple()calls()",
+        "no spaces()()"
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)